src/owners_file.rs

```diff
@@ -85,7 +91,18 @@ impl OwnersFileConfig {
                         line_number
                     )
                 })?;
-                Self::parse_text(config, &include_text, &include_path, repo_base.as_ref())?;
+
+                if seen_owners.contains_key(&include_path) {
+                    return Err(format_include_cycle_error(&include_path, &seen_owners));
+                }
```

Comment:
[optional] Since we're passing everything use in the if into the function anyways, we could easily more that into the function and have it be something like this instead:
```suggestion
                check_no_circular_include(&include_path, &seen_owners_files)?;
```
I'm okay either way.